### PR TITLE
WOR-1218: Update wb-libs, fix compilation errors and warnings

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/WOR-1218

## Summary of changes

@jsaun was trying to update wb-libs/workbench-azure dependency for https://github.com/broadinstitute/workbench-libs/pull/1494. However several of the wb-libs libraries were out of date, and updating them was causing breaking changes.

This is my attempt to update wb-libs to the latest and fix breaking changes. :) I also fixed a bunch of compiler warnings in tests, mostly related to implicits declared without a type.

### What

- Update each wb-libs module to the latest
- Fixed breaking changes related to fs2 `Files`
- Fixed compiler warnings, mostly in tests related to implicits not declaring types

### Why

- The impetus for this was to switch to using [admin credentials for AKS](https://github.com/broadinstitute/workbench-libs/pull/1494) due to a MSFT policy

### What to test

- Will rely on unit/automation tests

### Who tested and where

- [x] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [x] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala [hunk 1]
```diff
@@ -345,7 +346,8 @@ object Boot extends IOApp {
     logger: StructuredLogger[F],
     ec: ExecutionContext,
     as: ActorSystem,
-    F: Async[F]
+    F: Async[F],
+    files: Files[F]
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala:350]: `fs2` version bump requires the `Files` constraint if using `fs2.io` (which we are in a few places).

File: project/Dependencies.scala [hunk 2]
```diff
@@ -17,16 +17,16 @@ object Dependencies {
   val munitCatsEffectV = "1.0.7"
   val pact4sV = "0.9.0"
 
-  private val workbenchLibsHash = "63d2d78"
-  val serviceTestV = s"3.1-$workbenchLibsHash"
-  val workbenchModelV = s"0.18-$workbenchLibsHash"
+  private val workbenchLibsHash = "f87cad5"
```
Comment (rtitle) [inline project/Dependencies.scala:20]: This should be the latest and greatest wb-libs versions across all modules (except `google` which is marked with a TODO).

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala [hunk 3]
```diff
@@ -142,7 +143,7 @@ trait BillingProjectUtils extends LeonardoTestUtils {
 trait NewBillingProjectAndWorkspaceBeforeAndAfterAll extends BillingProjectUtils with BeforeAndAfterAll {
   this: TestSuite =>
 
-  implicit val ronTestersonAuthorization = Ron.authorization()
+  implicit val ronTestersonAuthorization: IO[Authorization] = Ron.authorization()
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:146]: All these changes are to fix the compiler warning:
```
Implicit definition should have explicit type
```
(seems like a reasonable request)

File: project/Dependencies.scala (hunk 2 above)
Comment (LizBaldo) [reply project/Dependencies.scala:20]: Ohh, I know what happened! I recently updated workbench lib, and then leo, but forgot to update this hash 🤦‍♀️ Sorry about that, and thanks for fixing it

**Review by LizBaldo — APPROVED** [review]
TYSM

**Comment by rtitle** [comment]
Just for sanity check, I deployed this to a BEE and made some GCP and Azure workspaces/runtimes/apps. So should be good I think once PR tests pass.

fiab-start failed, jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala (hunk 3 above)
Comment (Qi77Qi) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:146]: oh, I was appreciating ppl started doing this when reviewing PR...didn't realize this is compiler warning lol

**Review by Qi77Qi — APPROVED** [review]

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala (hunk 3 above)
Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala:146]: Lol yeah.. there were like 100 of them 🤦

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
1 swat failure:
```
org.broadinstitute.dsde.workbench.leonardo.runtimes.RuntimeDataprocSpec.should stop/start a Dataproc cluster with workers and preemptible workers

terra-quality-6658f4fc/automation-test-autps65wz errored due to List(RuntimeError(Required 'compute.images.get' permission for 'projects/broad-dsp-gcr-public/global/images/leo-dataproc-image-2-1-11-debian11-2023-07-28-21-05-58',Some(7),2023-08-25T14:30:45Z,Some(TraceId(1707dcf7802349719efc6249646b47da/3939911508519804487))))
```

I think this is a known flaky error. I'm going to go ahead and merge, given BEE testing also looked good (including creating a dataproc cluster).

**Pull request merged**